### PR TITLE
New test for checking the correct passing of the masses

### PR DESCRIPTION
**Pull request by Iximiel**
Like the title
I added a new test that checks that the masses are passed correclty.

There is a temporary commit that makes the PR download the target branch with the updated patch

### CODE REVIEW

**Comment by Iximiel** [comment]
I don't understand why the CI fails: it simply says that the exit code is '1' with no hints on what it may be the cause.

I tried it on my workstation, hoping in some useful error message, but it worked.

File: run_tests.sh
Comment (GiovanniBussi) [inline run_tests.sh:1]: @Iximiel I suppose the problem is with this script. I suggest that you try to push a commit where you reset the previous script (run_tests.sh). If that push does not fail, there's a mistake in the script, perhaps you used some non standard feature

Comment (Iximiel) [reply run_tests.sh:1]: If reverting the script works, I will be very confused, because locally the scripts works as I pushed it.

Comment (Iximiel) [reply run_tests.sh:1]: Ok, with some extra CI run I managed to make it work, I think the culprits was the `()` subshell that I'm used to use to not risk to forget the `cd ../` at the end of each for iteration

**Comment by Iximiel** [comment]
Now the PR is mergeable, and tests the main branch

**Pull request merged**